hack/make-rules/build_with_container.sh

```diff
@@ -23,7 +23,13 @@ set -o pipefail
 KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
 MOUNTPATH="${MOUNTPATH:-/kubeedge}"
 KUBEEDGE_BUILD_IMAGE=${KUBEEDGE_BUILD_IMAGE:-"kubeedge/build-tools"}
+DOCKER_GID="${DOCKER_GID:-$(grep '^docker:' /etc/group | cut -f3 -d:)}"
 
 set -x
-docker run --rm -v ${KUBEEDGE_ROOT}:${MOUNTPATH} -w ${MOUNTPATH} ${KUBEEDGE_BUILD_IMAGE} "$@"
+docker run --rm -u "${UID}:${DOCKER_GID}" \
+    --init \
+    --sig-proxy=false \
+    -e XDG_CACHE_HOME=/tmp/.cache \
+    -v ${KUBEEDGE_ROOT}:${MOUNTPATH} \
+    -w ${MOUNTPATH} ${KUBEEDGE_BUILD_IMAGE} "$@"
```

Comment:
-u: specify the current user to process the _output/local/bin directory permissions, or processes in containers will run as default user `root` and the directory will be assigned to the default root permission. FYI: https://medium.com/@mccode/understanding-how-uid-and-gid-work-in-docker-containers-c37a01d01cf
--init --sig-proxy=false:  process signals when pressing `ctrl+C` command
-e XDG_CACHE_HOME=/tmp/.cache: fix the error: `failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied`